README.md

```diff
@@ -77,53 +107,57 @@ Each numeric value reflect one of the following types:
  * `6` represents `gauge histogram`.
  * `7` represents `summary`.
 
-### name
+### source
 
-The name of the metric.
+The source of the metric in terms of where it originated.
 
- * Valid characters: `a-z,A-Z,0-9,_`.
- * Value is immutable.
+ * Valid value: `String` or `null`.
+ * Value is mutable.
+ * Defaults to `null`.
 
-### value
+### timestamp
 
-The value of the metric.
+A timestamp of when the metric was created.
 
- * Valid values: `Number` or `String`.
+ * Valid value: `Double` - epoc milliseconds.
  * Value is immutable.
+ * Defaults to `Date.now() / 1000`.
 
-### source
+### labels
 
-The source of the metric in terms of where it originated.
+A `Array` of labeled values. Each item in the `Array` must be an
+`Label` object. Please see the Labels section for further info.
 
- * Valid characters: `a-z,A-Z,0-9,_`.
- * Value is mutable.
+ * Valid value: `Array` of `Label` objects.
+ * Value is immutable.
+ * Defaults to an empty `Array`.
 
-### description
+### meta
 
-A human readable description of the metric.
+Available to be used to hold any data. The input is not validated.
 
- * Valid values: `String`.
+ * Valid value: `any`.
```

Comment:
Should meta be an object? Is it fine for it to be a number, string or an array?

Reply:
Initially I was thinking it can be anything since this is loosely informal data the consumer will use. Ex; for quantiles it could be just an Array with the values is. Though; an object might be a more structured way. How do you ex imagine the metadata for defining quantiles will look?

Reply:
I'm down with that.